MetaMask/metamask-extension
# Feat(MV3): Support Snaps in MV3

## **Description**
The work done in the proceeding offscreen PRs was done explicitly to hopefully make adopting the snaps executor written for offscreen support without much additional effort. This PR is not complete because it has a type error but sharing the code changes now.

This PR pulls forward the ProxySnapsExecutor from the snaps-execution-environments repository and initializes it with a runtime message stream. This allows the snaps executor to communicate through the offscreen document and enable support for snaps in MV3

## **Related issues**
[MetaMask-planning#1998](https://github.com/MetaMask/MetaMask-planning/issues/1998)

## **Manual testing steps**

1. Pull branch
2. run `yarn`
3. run `yarn start:mv3`
4. Go to https://metamask.github.io/snaps/test-snaps/latest/
5. Install some of the snaps by clicking "reconnect to X"
6. Interact with the buttons after installing snap and see they work.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
